### PR TITLE
dns cache: add dns_min_refresh_rate configuration

### PR DESCRIPTION
**Pull request by mattklein123**
Risk Level: Low
Testing: New UT
Docs Changes: Added
Release Notes: Added
Platform Specific Features: N/A

### CODE REVIEW

File: source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc [hunk 1]
```diff
@@ -353,8 +354,9 @@ void DnsCacheImpl::finishResolve(const std::string& host,
     }
     if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.use_dns_ttl")) {
       // Arbitrarily cap DNS re-resolution at 5s to avoid constant DNS queries.
```
Comment (goaway) [inline source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc:356]: Looks like you may want to update this comment as well.

**Comment by goaway** [comment]
This looks good from the EM side, thank you @mattklein123!

File: source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc (hunk 1 above)
Comment (mattklein123) [reply source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc:356]: Oops yup thanks will fix.

File: source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc [hunk 2]
```diff
@@ -352,9 +353,10 @@ void DnsCacheImpl::finishResolve(const std::string& host,
       addCacheEntry(host, new_address, address_list, response.front().addrInfo().ttl_);
     }
     if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.use_dns_ttl")) {
-      // Arbitrarily cap DNS re-resolution at 5s to avoid constant DNS queries.
-      dns_ttl =
-          std::max<std::chrono::seconds>(std::chrono::seconds(5), response.front().addrInfo().ttl_);
+      // Arbitrarily cap DNS re-resolution at min_refresh_interval_ to avoid constant DNS queries.
```
Comment (alyssawilk) [inline source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc:356]: have you considered changing your TTL if you don't like it? :-P

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**